onnxruntime/contrib_ops/cuda/quantization/matmul_nbits.cu

```diff
@@ -192,7 +192,7 @@ __global__ void __launch_bounds__(kWarpSize* kColsPerThreadBlock) MatMulFloatInt
   const int lane_id = threadIdx.x;
   const int warp_id = WarpUniform(threadIdx.y);
   const int n_id = n_block_id * kColsPerThreadBlock + warp_id;
-  constexpr int k_per_iter = 256;
+  constexpr int k_per_iter = kWarpSize * kColsPerThreadBlock;
 
```

Comment:
k_per_iter is not computed by kWarpSize * kColsPerThreadBlock. It is computed from kWarpSize * kElementsPerThreadPerIteration, in which kElementsPerThreadPerIteration is 8. You may define a constant value kElementsPerThreadPerIteration.